### ISSUE TITLE
Get label-content-name-mismatch consistent with ACT's equivalent

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#5203]
Axe-core is currently set to partially consistent for this ACT rule:

https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/

Part of that comes from us not having updated in a while. That needs to be sorted. But after sorting that failed example 4 isn't failed by Axe-core.

**Comment by chutchins25** [comment]
Cross-referencing #4678, which includes a name-from-content case axe currently doesn't evaluate at all:

```html
<a href="#">deque <span class="visually-hidden">labs</span> are great</a>
```

With no `aria-label`/`aria-labelledby`, `label-content-name-mismatch-matches` bails, so axe never fails it — even though the visible label ("deque are great") isn't contained in the accessible name ("deque labs are great", since the SR-visible hidden text is part of the name). That's the same shape as "failed example 4 isn't failed by Axe-core": axe's applicability is narrower than 2ee8b8. Flagging it here so it's covered when the rule is brought in line with the ACT rule.

**Comment by straker** [comment]
`<a href="#">deque <span class="visually-hidden">labs</span> are great</a>` should never fail according to the ACT rule. The rule's applicability section states (emphasis mine):

> This rule applies to any element for which all the following is true:

> * The element has a [semantic role](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#semantic-role) that is a [widget](https://www.w3.org/TR/wai-aria-1.2/#widget_roles) that [supports name from content](https://www.w3.org/TR/wai-aria-1.2/#namefromcontent); and
> * The element has [visible text content](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#visible-text-content); and
> * __The element has an aria-label or aria-labelledby attribute; and__
> * Neither the element’s accessible name nor its visible label contains any abbreviations; and
> * Every word that appears in both the element’s accessible name and its visible label uses the same spelling and hyphenation in both places.

Since the element does not have an `aria-label(ledby)` attribute we don't run the rule against it. So the example provided should just be an ignored case rather than something we need to cover. The matcher for the rule already has tests to cover missing attributes so I don't think there's anything we need to do to handle this.

**Comment by straker** [comment]
Just to be clear, getting consistent means our act test passes on the new examples. This means implementing the new algorithm as defined in https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#label-in-name-algorithm. Our current code just does a substring comparison of the accessible name and the visible text. We'll need to add the other steps to that part of the code.

We are accepting that Inapplicable 5 & 6 will fail instead of incomplete and should be skipped. See https://github.com/act-rules/act-rules.github.io/pull/2443

**Comment by chutchins25** [comment]
Mapping what's left here, since #5345 landed and the remaining scope isn't obvious from the sibling issues.

Taking your definition — *"getting consistent means our act test passes on the new examples"* — the state against `wcag-act-rules#main` is **35 passing / 3 pending / 0 failing**, and two of the three pending are the divergence you accepted (`Inapplicable Example 5` and `6`, act-rules/act-rules.github.io#2443).

**To close:**

- [x] Word-level contiguous-subsequence matching — #5345
- [ ] Remove parenthesised content — #5360 (in review; unskips `Passed Example 14`)
- [ ] `Passed Example 11` — #5207. `visibleVirtual` trims whitespace-only inline elements, so `Download` and `specification` join into one token. Waiting on your `visibleVirtual` call in the comment above.
- [ ] Bump `wcag-act-rules` off the pinned SHA — #5346. PR CI currently runs 15 2ee8b8 testcases while the nightly runs 38, so "passes on the new examples" isn't demonstrable from PR CI until this lands.

**Not blocking:** #5308 (CJK / full-width punctuation) is the algorithm's NFKD step and language-aware word segmentation is its last, but no 2ee8b8 example exercises either, so neither gates this issue. Segmentation isn't tracked anywhere — say if you'd rather it had one.

Correct me if any of that misreads what "consistent" covers.
